### PR TITLE
update to 0.9.1: don't try to run removed tests; re-add 1.5 tests (one only in minimal); remove bls_verify_multiple(...) and AttestationDataAndCustodyBit; and update process_attester_slashing(...), get_indexed_attestation(...), and is_valid_indexed_attestation(...)

### PR DESCRIPTION
**Pull request by tersec**
https://github.com/ethereum/eth2.0-specs/releases/tag/v0.9.1

Mostly just a straightforward simplification. One hitch is that in the block sanity test fixture, the attester slashing test fails and was disabled, as near as I can tell because either SSZ reading or Merkle hashing disagrees with the test case's post-result. There are no other discrepancies in that test's result (e.g., slashing amounts or slashed/rewarded indices), only the `latest_block_header.body_root` field, which is simply the `hash_tree_root(blck.body)` per https://github.com/ethereum/eth2.0-specs/blob/v0.9.1/specs/core/0_beacon-chain.md#block-header and https://github.com/status-im/nim-beacon-chain/blob/7197d6c70388fcc8e9055f45ffa329009cb1d167/beacon_chain/spec/state_transition_block.nim#L68-L77:
```
test case beginning
blck.body hash = (data: [153, 10, 109, 198, 89, 156, 215, 217, 181, 114, 99, 188, 169, 125, 185, 120, 209, 79, 170, 192, 120, 18, 146, 129, 116, 18, 248, 249, 183, 29, 108, 229])
false
Diff: sr_pre`gensym2996837.latest_block_header.body_root.data[0] = 153
and   sr_post`gensym2996838.latest_block_header.body_root.data[0] = 165

false
Diff: sr_pre`gensym2996837.latest_block_header.body_root.data[1] = 10
and   sr_post`gensym2996838.latest_block_header.body_root.data[1] = 190

false
Diff: sr_pre`gensym2996837.latest_block_header.body_root.data[2] = 109
and   sr_post`gensym2996838.latest_block_header.body_root.data[2] = 129

false
Diff: sr_pre`gensym2996837.latest_block_header.body_root.data[3] = 198
and   sr_post`gensym2996838.latest_block_header.body_root.data[3] = 12

false
Diff: sr_pre`gensym2996837.latest_block_header.body_root.data[4] = 89
and   sr_post`gensym2996838.latest_block_header.body_root.data[4] = 0

false
Diff: sr_pre`gensym2996837.latest_block_header.body_root.data[5] = 156
and   sr_post`gensym2996838.latest_block_header.body_root.data[5] = 43

false
Diff: sr_pre`gensym2996837.latest_block_header.body_root.data[6] = 215
and   sr_post`gensym2996838.latest_block_header.body_root.data[6] = 246

false
Diff: sr_pre`gensym2996837.latest_block_header.body_root.data[7] = 217
and   sr_post`gensym2996838.latest_block_header.body_root.data[7] = 159

false
Diff: sr_pre`gensym2996837.latest_block_header.body_root.data[8] = 181
and   sr_post`gensym2996838.latest_block_header.body_root.data[8] = 53

false
Diff: sr_pre`gensym2996837.latest_block_header.body_root.data[10] = 99
and   sr_post`gensym2996838.latest_block_header.body_root.data[10] = 134

false
Diff: sr_pre`gensym2996837.latest_block_header.body_root.data[11] = 188
and   sr_post`gensym2996838.latest_block_header.body_root.data[11] = 168

false
Diff: sr_pre`gensym2996837.latest_block_header.body_root.data[12] = 169
and   sr_post`gensym2996838.latest_block_header.body_root.data[12] = 119

... etc, through latest_block_header.body_root.data[31]
```
The tests in question are https://github.com/ethereum/eth2.0-spec-tests/tree/v0.9.1/tests/minimal/phase0/sanity/blocks/pyspec_tests/attester_slashing

### CODE REVIEW

File: beacon_chain/spec/crypto.nim [hunk 1]
```diff
@@ -178,35 +178,16 @@ func bls_verify*(
     if sig.kind != Real or pubkey.kind != Real:
       # TODO: chronicles warning
       return false
+    # TODO bls_verify_multiple(...) used to have this workaround, and now it
+    # lives here. No matter the signature, there's also no meaningful way to
+    # verify it -- it's a kind of vacuous truth. No pubkey/sig pairs.
+    if pubkey == ValidatorPubKey():
```
Comment (mratsim) [inline beacon_chain/spec/crypto.nim]: I think `default(ValidatorPubKey)` would be more readable (thinking of future audits there)

File: beacon_chain/spec/state_transition_block.nim [hunk 2]
```diff
@@ -244,12 +244,11 @@ proc process_attester_slashing*(
...
     ## is compelling perf evidence otherwise.
-    let
-      attesting_indices_1 = attestation_1.custody_bit_0_indices & attestation_1.custody_bit_1_indices
-      attesting_indices_2 = attestation_2.custody_bit_0_indices & attestation_2.custody_bit_1_indices
-    for index in sorted(toSeq(intersection(toSet(attesting_indices_1),
-        toSet(attesting_indices_2)).items), system.cmp):
-      if is_slashable_validator(state.validators[index.int], get_current_epoch(state)):
+    for index in sorted(toSeq(intersection(
+        toSet(attestation_1.attesting_indices),
+        toSet(attestation_2.attesting_indices)).items), system.cmp):
```
Comment (mratsim) [inline beacon_chain/spec/state_transition_block.nim]: `toSet` is deprecated, `toHashSet` is the new recommended way

File: beacon_chain/spec/crypto.nim (hunk 1 above)
Comment (tersec) [reply beacon_chain/spec/crypto.nim]: https://github.com/status-im/nim-beacon-chain/pull/544/commits/53c635a17c63e384580102782efc8b0082dbbc4f

File: beacon_chain/spec/state_transition_block.nim (hunk 2 above)
Comment (tersec) [reply beacon_chain/spec/state_transition_block.nim]: https://github.com/status-im/nim-beacon-chain/pull/544/commits/53c635a17c63e384580102782efc8b0082dbbc4f

**Pull request merged**